onnxruntime/hosting/request_handler.h

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "http_server.h"
+#include "json_handling.h"
+
+namespace onnxruntime {
+namespace hosting {
+
+namespace beast = boost::beast;
+namespace http = beast::http;
+
+void BadRequest(HttpContext& context, const std::string& error_message) {
+  http::response<http::string_body> res{http::status::bad_request, context.request.version()};
+  res.set(http::field::server, BOOST_BEAST_VERSION_STRING);
+  res.set(http::field::content_type, "text/html");
```

Comment:
We better define a "error_detail" structure like `{"error_code": xxx, "error_message: yyy}` and return `application/json` to provide enough information.  

Also, `request_id` should be add to the header as well. (do not have to be in this pr)

Reply:
I would like to move everything beast + header related eventually to the context class. But for now I've created a quick json error